AutoRouteManager.php

```diff
@@ -64,22 +71,45 @@ public function __construct(
      */
     public function buildUriContextCollection(UriContextCollection $uriContextCollection)
     {
-        $this->getUriContextsForDocument($uriContextCollection);
+        $this->collectionBuilder->build($uriContextCollection);
 
         foreach ($uriContextCollection->getUriContexts() as $uriContext) {
-            $existingRoute = $this->adapter->findRouteForUri($uriContext->getUri(), $uriContext);
+            $translatedSubject = null;
+            $subject = $uriContextCollection->getSubjectObject();
 
-            $autoRoute = null;
+            if (null !== $uriContext->getLocale()) {
+                $translatedSubject = $this->adapter->translateObject($subject, $uriContext->getLocale());
+            }
+
+            if (null === $translatedSubject) {
+                @trigger_error('AdapterInterface::translateObject() has to return the subject as of version 1.1, support for by reference will be removed in 2.0.', E_USER_DEPRECATED);
+            } else {
+                if ($translatedSubject !== $subject) {
+                    $uriContext->setTranslatedSubjectObject($translatedSubject);
```

Comment:
The `UriContext` now contains both the `subject` and the `translatedSubject`. This is necessary since we now imply that the translated and subject object may be different object instances - the consumer may need to access the object ID of the original object f.e.